test/integration/simple_operations_test.lua

```diff
@@ -391,3 +391,114 @@ pgroup:add('test_upsert', function(g)
     t.assert_equals(err, nil)
     t.assert_equals(result.rows, {{67, 1143, 'Mikhail Saltykov-Shchedrin', 63}})
 end)
+
+pgroup:add('test_object_with_nullable_fields', function(g)
+    -- Insert
+    local result, err = g.cluster.main_server.net_box:call(
+        'crud.insert_object', {'tags', {id = 1, is_green = true}})
+    t.assert_equals(err, nil)
+
+    -- {1, 477, NULL, true, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL}
+    local objects = crud.unflatten_rows(result.rows, result.metadata)
+    t.assert_equals(objects, {
+        {
+            bucket_id = 477,
+            id = 1,
+            is_blue = box.NULL,
+            is_correct = box.NULL,
+            is_dirty = box.NULL,
+            is_green = true,
+            is_long = box.NULL,
+            is_red = box.NULL,
+            is_short = box.NULL,
+            is_sweet = box.NULL,
+            is_useful = box.NULL,
+            is_yellow = box.NULL,
+        }
+    })
+
+    -- Update
+    -- {1, 477, NULL, true, NULL, NULL, true, NULL, NULL, NULL, NULL, NULL}
+    -- Shouldn't failed because of https://github.com/tarantool/tarantool/issues/3378
+    result, err = g.cluster.main_server.net_box:call(
+        'crud.update', {'tags', 1, {{'=', 'is_sweet', true}}})
+    t.assert_equals(err, nil)
+    objects = crud.unflatten_rows(result.rows, result.metadata)
+    t.assert_equals(objects, {
+        {
+            bucket_id = 477,
+            id = 1,
+            is_blue = box.NULL,
+            is_correct = box.NULL,
+            is_dirty = box.NULL,
+            is_green = true,
+            is_long = box.NULL,
+            is_red = box.NULL,
+            is_short = box.NULL,
+            is_sweet = true,
+            is_useful = box.NULL,
+            is_yellow = box.NULL,
+        }
+    })
+
+    -- Replace
+    -- {2, 401, NULL, true, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL}
+    result, err = g.cluster.main_server.net_box:call(
+        'crud.replace_object', {'tags', {id = 2, is_green = true}})
+    t.assert_equals(err, nil)
+    objects = crud.unflatten_rows(result.rows, result.metadata)
+    t.assert_equals(objects, {
+        {
+            bucket_id = 401,
+            id = 2,
+            is_blue = box.NULL,
+            is_correct = box.NULL,
+            is_dirty = box.NULL,
+            is_green = true,
+            is_long = box.NULL,
+            is_red = box.NULL,
+            is_short = box.NULL,
+            is_sweet = box.NULL,
+            is_useful = box.NULL,
+            is_yellow = box.NULL,
+        }
+    })
+
+    local _
+    -- Upsert: first is insert then update
+    -- {3, 2804, NULL, NULL, NULL, NULL, NULL, true, NULL, NULL, NULL, NULL}
+    _, err = g.cluster.main_server.net_box:call(
```

Comment:
Why not `local _, err`?

Reply:
AFAIU our linter settings is quite dehumanized and prohibits variable redefinition.

Reply:
No, it allows variable redefinition and it is used everywhere in tests.

Reply:
Well, fixed